src/services/codefixes/disableJsDiagnostics.ts

```diff
@@ -35,15 +35,21 @@ namespace ts.codefix {
                 fixId: undefined,
             }];
         },
-        fixIds: [fixId], // No point applying as a group, doing it once will fix all errors
-        getAllCodeActions: context => codeFixAllWithTextChanges(context, errorCodes, (changes, err) => {
-            if (err.start !== undefined) {
-                changes.push(getIgnoreCommentLocationForLocation(err.file!, err.start, getNewLineOrDefaultFromHost(context.host, context.formatContext.options)));
-            }
-        }),
+        fixIds: [fixId],
+        getAllCodeActions: context => {
+            const seenLines = createMap<true>(); // Only need to add `// @ts-ignore` for a line once.
+            return codeFixAllWithTextChanges(context, errorCodes, (changes, err) => {
+                if (err.start !== undefined) {
+                    const { line, change } = getIgnoreCommentLocationForLocation(err.file!, err.start, getNewLineOrDefaultFromHost(context.host, context.formatContext.options));
+                    if (addToSeen(seenLines, line)) {
+                        changes.push(change);
+                    }
+                }
+            });
+        },
     });
 
-    function getIgnoreCommentLocationForLocation(sourceFile: SourceFile, position: number, newLineCharacter: string): TextChange {
+    function getIgnoreCommentLocationForLocation(sourceFile: SourceFile, position: number, newLineCharacter: string): { line: number, change: TextChange } {

```

Comment:
Unless there's an existing convention, I would be tempted to rename "line" to "lineNumber".  At the call site, I couldn't figure out whether it was the line text or the line number.  (Yes, I realize that using the line text would make no sense.)